File: 2233-maximum-product-after-k-increments/2233-maximum-product-after-k-increments.py
class Solution:
    def maximumProduct(self, nums: List[int], k: int) -> int:
        heapify(nums)
        
        for _ in range(k):
            heappush(nums, heappop(nums) + 1)
        
        MOD, ans = 1_000_000_007, 1
        
        for num in nums:
            ans = (ans * num) % MOD
            
        return ans
        

File: 2233-maximum-product-after-k-increments/2233-maximum-product-after-k-increments.cpp
class Solution {
public:
    int maximumProduct(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> pq;
        
        for (int num : nums) pq.push(num);
        
        for (int i = 0; i < k; ++i){
            int num = pq.top(); pq.pop();
            pq.push(num + 1);
        }
        
        constexpr int MOD = 1'000'000'007;
        long long ans = 1;
        while (!pq.empty()){
            ans = (ans * pq.top()) % MOD;
            pq.pop();
        }
        
        return ans;
    }
};